packages/data-api/src/AnalyticsFetchQuery.js

```diff
@@ -3,124 +3,172 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
-import { DataFetchQuery, ANSWER_SPECIFIC_FIELDS } from './DataFetchQuery';
+import { SqlQuery } from './SqlQuery';
+
+const VALUE_AGGREGATION_FUNCTIONS = {
+  SUM: 'sum(value::numeric)::text',
+  MOST_RECENT: 'most_recent(value, date)',
+};
+
+const COMMONLY_SUPPORTED_CONFIG_KEYS = [
+  'dataSourceEntityType',
+  'dataSourceEntityFilter',
+  'orgUnitMap',
+];
 
 const AGGREGATION_SWITCHES = {
   FINAL_EACH_DAY: {
     groupByPeriodField: 'day_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   FINAL_EACH_WEEK: {
     groupByPeriodField: 'week_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
+  },
+  SUM_EACH_WEEK: {
+    groupByPeriodField: 'week_period',
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.SUM,
   },
   FINAL_EACH_MONTH: {
     groupByPeriodField: 'month_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   FINAL_EACH_YEAR: {
     groupByPeriodField: 'year_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   MOST_RECENT: {
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   MOST_RECENT_PER_ORG_GROUP: {
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
     aggregateEntities: true,
+    supportedConfigKeys: ['aggregationEntityType', 'orgUnitMap'],
   },
   SUM_PER_ORG_GROUP: {
-    sum: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.SUM,
     aggregateEntities: true,
+    supportedConfigKeys: ['aggregationEntityType', 'orgUnitMap'],
   },
   SUM_PER_PERIOD_PER_ORG_GROUP: {
-    sum: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.SUM,
     aggregateEntities: true,
-    groupByPeriodField: 'day_period', // can assume first internal aggregation period type is daily
+    groupByPeriodField: 'period',
+    supportedConfigKeys: ['aggregationEntityType', 'orgUnitMap'],
   },
 };
 
-export class AnalyticsFetchQuery extends DataFetchQuery {
+const supportsConfig = (aggregationSwitch, config) => {
+  if (!config) {
+    return true;
+  }
+
+  const switchSupportedKeys = aggregationSwitch.supportedConfigKeys || [];
+  return Object.keys(config).every(
+    key => COMMONLY_SUPPORTED_CONFIG_KEYS.includes(key) || switchSupportedKeys.includes(key),
+  );
+};
+
+export class AnalyticsFetchQuery {
```

Comment:
Decided to _yet again_ refactor the structure of this class. Now that the response type has deviated from the `EventsFetchQuery` I felt it would be best to decouple the two by removing the inheritance model and either shifting logic to `SqlQuery` or just duplicating between the two.